src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1473,13 +1470,34 @@
   <data name="AvoidAssemblyGetFilesInSingleFileMessage" xml:space="preserve">
     <value>'{0}' will throw for assemblies embedded in a single-file app</value>
   </data>
-  <data name="PlatformCompatibilityCheckUnsupportedOsMessage" xml:space="preserve">
-    <value>'{0}' is unsupported on '{1}'</value>
+  <data name="PlatformCompatibilityUnsupportedMessage" xml:space="preserve">
+    <value>'{0}' is unsupported on: {1}</value>
   </data>
-  <data name="PlatformCompatibilityCheckSupportedOsMessage" xml:space="preserve">
-    <value>'{0}' is supported on '{1}'</value>
+  <data name="PlatformCompatibilityOnlySupportedMessage" xml:space="preserve">
+    <value>'{0}' is only supported on: {1}</value>
   </data>
   <data name="PreferStringContainsOverIndexOfCodeFixTitle" xml:space="preserve">
     <value>Replace with 'string.Contains'</value>
   </data>
+  <data name="CallSiteCrossPlatformSuffix" xml:space="preserve">
+    <value>but this call site is reachable on all platforms.</value>
+  </data>
+  <data name="CallSiteReachableOnPlatformSuffix" xml:space="preserve">
+    <value>but this call site is reachable on {0}.</value>
+  </data>
+  <data name="CallSiteUnreachableOnPlatformSuffix" xml:space="preserve">
+    <value>but this call site is unreachable on {0}.</value>
+  </data>
+  <data name="PlatformFromVersionAndLaterSuffix" xml:space="preserve">
+    <value>'{0}' from version {1} and later, </value>
+  </data>
+  <data name="PlatformFromVersionToVersionSuffix" xml:space="preserve">
+    <value>'{0}' from version {1} to {2}, </value>
+  </data>
+  <data name="PlatformVersionAndLaterSuffix" xml:space="preserve">
+    <value>'{0}' {1} and later, </value>
+  </data>
+  <data name="PlatformVersionAndBeforeSuffix" xml:space="preserve">
+    <value>'{0}' {1} and before, </value>
```

Comment:
Was need to construct messages from several parts, might not good for localization, for using predefined strings for all possible combinations will need to add many more rules ...

Reply:
Yes, I have been told multiple times by LOC team that combining different resource strings to form the final string is _never_ a good solution. I don't know of a better alternative though.

Reply:
The guidance I've collected over the years for this scenario is:

- Never concatenate strings together; always have a single message since sentence structures are not consistent across languages
- Consider using bullets instead of lists in a sentence
- Consider producing multiple resource strings to represent different scenarios and programmatically choose the correct resource string; such as:
    - When there is a single item in the list
    - When there are two items in the list
    - When there are three or more items in the list

I don't think it's going to surface here, but a related tip worth sharing is that we can never assume quantities are as simple as having a singular and plural form.

- Some languages (e.g. Russian) have different words not just for singular and plural, but for 0, 1, 2, 3, 4, and 5 of something
- Therefore you cannot assume messages like "{0} platform" or "{1} platforms" will suffice
- Instead, favor using a label and data combo such as "Number of platforms: {0}"
- But if natural language is preferred, then utilize specific resource strings to capture the most common quantities (One, Two, Three, Four, Five) and a label/data combo message as the catch-all

Reply:
Ok, then a separate descriptor for each message is preferred, it would be great to get dom feedback for the messages produced. I have added them as a comment to the test cases for review

Reply:
I'm not sure if you've already made changes here based on the feedback above, but to expand on my notes above:

- Do not use one resource string for what the API is annotated with and another resource string for what the call site supports; instead, you need unique resource strings for each combination such as `ApiSupportedOnPlatformsCallSupportedOnAllPlatforms`
- There should be zero string concatenation to build a message (no StringBuilders)
- You can however use nested resources where you get one parameterized message and use that as a value inside another parameterized message -- such as having a message that represents a list of platforms, getting one value that is the list of supported platforms and another value that is the list of reachable platforms, and using those as placeholders inside the full message
- Assuming you use `String.Join` to create the string that represents the list of platforms, the item separator string also needs to be localizable.  In English, it would be `", "`, but there will be some languages that use a different separator string.

I also recommend structuring the overall message into 2 sentences in English (but still as one message):

`'{API}' is supported on: {supportedPlatformsList}. This call site is reachable on: {reachablePlatformsList}.`

This approach will avoid trouble with commas being used both in the platforms lists and within the sentence structure itself; and different languages would then be able to deal with this in targeted ways.

Reply:
> Assuming you use String.Join to create the string that represents the list of platforms, the item separator string also needs to be localizable. In English, it would be ", ", but there will be some languages that use a different separator string.

Interesting, thanks 

> '{API}' is supported on: {supportedPlatformsList}. This call site is reachable on: {reachablePlatformsList}.

The {supportedPlatformsList} and {reachablePlatformsList} still would need to use `'{platform}' {version} and before`, in case any of platforms versioned